### DISCUSSION TITLE
LAB6 UART simulation

### DISCUSSION THREAD

**Discussion by ChenChung107360206** [bol-edu/HLS-SOC-Discussions#180] [Q&A]
我們把firmware跟硬體整合起來，但uart在tx結束後就會卡住，不會出現rx，想請問有沒有人遇過類似的問題
![image](https://github.com/bol-edu/HLS-SOC-Discussions/assets/43467000/51cae719-ceff-4294-ae1d-903d0558217d)

**Comment by TKAndyLiang** [comment]
你可以檢查看看你的run_sim裡面有沒有define到 USER_PROJ_IRQ0_EN 這個參數，可以參考uart那個testbench裡面的run_sim。

**Comment by JIAJIA4** [comment]
同學你好，我有遇過類似的問題，當初是因為我的uart_tb.v的$finish;太早了，導致還沒出現rx就結束。

**Comment by chickenchennnn** [comment]
如果你tb裡面單純是以send_data_task完後就finish的話，你並不會等到整個uart loop back結束才finish，自然也不會看到rx的訊息。

**Comment by ChenChung107360206** [comment]
感謝各位的回覆!
